eng/ci/templates/official/jobs/build-artifacts.yml

```diff
@@ -0,0 +1,193 @@
+jobs:
+- job: BuildArtifacts
+  displayName: Build Artifacts
+
+  templateContext:
+    outputParentDirectory: $(Build.ArtifactStagingDirectory)
+    outputs:
+    - output: pipelineArtifact
+      displayName: Publish nuget packages
+      path: $(Build.ArtifactStagingDirectory)/NugetPackages
+      artifact: NugetPackages
+
+  pool:
+    name: 1es-pool-azfunc
+    image: 1es-windows-2022
+    os: windows
+
+  variables:
+    buildNumber: $(Build.BuildNumber)
```

Comment:
You will need to copy the build number logic from our existing CI files, as we want this to be empty string on release builds.